sklearn/ensemble/bagging.py

```diff
@@ -52,6 +53,11 @@ def _parallel_build_estimators(n_estimators, ensemble, X, y, sample_weight,
     bootstrap_features = ensemble.bootstrap_features
     support_sample_weight = has_fit_parameter(ensemble.base_estimator_,
                                               "sample_weight")
+    # Logistic regression does not support sample weights with liblinear
+    if (isinstance(ensemble.base_estimator_,LogisticRegression) and 
```

Comment:
Regarding your numbered list of options: 1 and 2 are most practical, as you mentioned. I don't think 3 is worth creating for such a specific case, while 4 already exists but has stalled for about a year (see #2784 - thanks to @MechCoder for picking up recently).

I don't think the `try` / `except` would be quite the same as option 1 since the existence of sample weights actually determines how the model is fit. The logic in `_parallel_build_estimators` is as follows:
```
if sample weights are supported:
   do A
if not:
   do B
```
I think keeping this in tact is the better option. What do you think?